datalad/core/local/save.py

```diff
@@ -267,6 +284,17 @@ def __call__(path=None, message=None, dataset=None,
                             type='dataset')
                 paths_by_ds[superds] = superds_status
 
+        # We need to anticipate empty (amend) commits in subdatasets that
+        # currently appear to be clean. Change state of any subdataset to
+        # modified in order for it to be committed upwards after amending.
+        # Note, that at least a message is required in order to actually commit
+        # in a subdataset that otherwise is clean.
+        if amend and recursive and message:
+            for d in paths_by_ds:
+                for p in paths_by_ds[d]:
+                    if paths_by_ds[d][p]['type'] == 'dataset':
+                        paths_by_ds[d][p]['state'] = 'modified'
```

Comment:
Hmm, amending recursively seems messsy.  As a naive user, I think I'd expect it to only amend subdatasets that changed in the last commit.  Perhaps `--amend` and `--recursive` should just be marked as incompatible, at least initially.  If I'm thinking about gh-5412 correctly, recursive amending isn't needed there.

Reply:
Not needed there indeed.

> As a naive user, I think I'd expect it to only amend subdatasets that changed in the last commit.

I think that expectation isn't naive at all. ;)
I'm not sure what any user would expect, though. Not even sure what I want it to be. ;-)
Fulfilling your expectation instead seems perfectly reasonable, since the only "real" usecase I can think of is a previous `save --recursive`.

Would like to hear other @datalad/developers view on this.

Reply:
Okay, so I think, we should go with @kyleam's expectation: On `save --amend --recursive`, amend only in subdatasets, that were changed in the previous commit. Modifications in other subdatasets should lead to a separate commit in them, but the changed commit should then be amended upwards. This seems to be the most intuitive thing to aim for, in my mind. I expect it to be a mess implementation-wise, though. Will have a look.

Reply:
Moved to #5455. Making `--amend` and `--recursive` incompatible for now.